File: exercicios-lp/Python/metodo-de-julia.py
#!/usr/bin/python3.13
import clear
clear

# Criar um programa que mostre uma raiz quadrada pelo método da Regressão de Júlia.

raiz = int(input(" Informe a raiz que você quer procurar: "))
w = int(input(" Chute um candidato para essa raiz: "))
candidato = w + 1
w2 = w * w
test = 0
i=0

while (test != raiz):
    
    test = w2 + w + candidato
    if (test == raiz):
        print(" A raiz é:",candidato)
        break
    
    print(i)
    i +=1
    w2 = test
    w = candidato
    candidato +=1

    if (test > raiz):
        print(" \nA raíz é irracional, paciência! ")
        break
  

File: exercicios-lp/MetodoDeJulia.java
import java.util.Scanner;

public class MetodoDeJulia {
  
  public static void main (String[]args) {
    clear.limpar();
    int radicando, palpite, candidato, w, teste;
    Scanner entrada = new Scanner(System.in);

    System.out.println(" Digite o valor do radicando: ");
    radicando = entrada.nextInt();

    System.out.println(" Digite o palpite: ");
    palpite = entrada.nextInt();
    teste = 0;

    candidato = palpite + 1;
    w = palpite * palpite;

    while (teste != radicando){
      
      System.out.println("teste = " + teste);

      teste = w + palpite + candidato;
      if (teste == radicando){
        System.out.println(" A raiz é: " + candidato);
      }
      if (teste > radicando){
        System.out.println(" A raiz é irracional, paciência! ");
        break;
      }
      
      w = teste;
      palpite = candidato;
      candidato += 1;

    }
    entrada.close();
    
  }

}


